hcl2template/types.packer_config.go

```diff
@@ -489,6 +497,8 @@ func (cfg *PackerConfig) getCoreBuildProvisioner(source SourceUseBlock, pb *Prov
 		return packer.CoreBuildProvisioner{}, diags
 	}
 
+	flatProvisionerCfg, _ := decodeHCL2Spec(pb.HCL2Ref.Rest, ectx, provisioner)
```

Comment:
Lets move this initialization closer to where it is used which right about line 521. 

That said, do we want to capture details about the provisioner being wrapped?

Reply:
Theoretically it should yes! I'll test it to be sure though

Reply:
Oh I need to test this.